awscli/customizations/s3/results.py

```diff
@@ -186,7 +195,10 @@ def _get_ongoing_dict_key(self, result):
                 'Any result using _get_ongoing_dict_key must subclass from '
                 'BaseResult. Provided result is of type: %s' % type(result)
             )
-        return ':'.join([result.transfer_type, result.src, result.dest])
+        key_components = [result.transfer_type, result.src]
```

Comment:
What about just `':'.join(str(el) for el in [...])` to avoid the `dest is None` special casing?